### ISSUE TITLE
Model Handler and model Trainer refactor

### ISSUE THREAD

**Issue by leoglonz** [mhpi/generic_deltaModel#6]
### Summary

General trainer attributes need to be removed from Model Handler and added to Trainer to adhere to a separation of concerns. Model Handler serves as a gateway to 𝛿Models, and also BMI, and as such, would be critically positioned to be a container for methods designed to act on 𝛿Models.

### Actions

- Remove optimizer and loss function init/handling from Model Handler. This should instead exist in Trainer files for general training tasks.
- Refine Model Handler to a simple wrapper around 𝛿Models. 
- Design in the capability to receive methods that act on 𝛿Models (e.g., variational DA, multiscale, ensembling)

### Considerations

- Is it right and possible to make Model Handler a container for methods?

*Assignments*: @JiangtaoLiud, @leoglonz
